pkg/api/controller.go

```diff
@@ -2113,7 +2241,9 @@ func (c *Controller) handleAPIErrorCallback(ctx context.Context, w http.Response
 	case errors.Is(err, graveler.ErrTooManyTries):
 		log.Debug("Retried too many times")
 		cb(w, r, http.StatusLocked, "Too many attempts, try again later")
-
+	case errors.Is(err, graveler.ErrPreconditionFailed):
```

Comment:
What changed that we need this now?  Or is it a quick fix for a bug that you discovered along the way?

Reply:
This error is now returned from Graveler when the conditional update of branch protection rules fails.